### PR TITLE
fix: match native transfer state cleanup

### PR DESCRIPTION
**Pull request by HashimTheArab**
## Summary
- track and clear backend entities, boss bars, player-list entries, and scoreboard objectives during native transfers
- reset sounds, weather, difficulty, rotation, game mode, and game rules from destination game data
- complete runtime/unique ID collision handling and suppress duplicate backend self-spawns

## Verification
- `go test -race ./...`
- `go vet ./...`
- `go test ./... && go vet ./...` in `example/default`
- `go test ./... && go vet ./...` in `example/dragonfly`
- `git diff --check`

---

> [!NOTE]
> **Medium Risk**
> Changes the hot path for every server→client packet and transfer synchronization; incorrect rewriting or cleanup could cause ghost entities, broken interactions, or inconsistent client state after transfers.
> 
> **Overview**
> Aligns the native RakNet proxy’s **backend transfer** path with fuller client reset behavior and safer entity ID rewriting.
> 
> A new **`backendStateTracker`** records client-visible state from forwarded server packets (spawned entities, boss bars, tab-list players, scoreboard objectives) and **`resetTransferState`** now flushes matching teardown packets before the existing mob-effect and dimension reset sequence. **`transferResetPackets`** additionally stops all sounds and weather, restores **difficulty**, and resets **pitch/yaw** on `MovePlayer`.
> 
> **`rewriteServerPacket`** now returns whether to forward the packet: backend **self** `AddActor`/`AddPlayer` spawns are dropped, and runtime/unique ID rewrites use a **`math.MaxInt64` sentinel** when backend IDs would collide with the stable client identity (with inverse handling on client `Interact` / use-on-entity transactions). Coverage is extended to more entity-related packet types.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit c86550c447bb6efa724cd0a962f0d49606f9930c. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

## Summary by CodeRabbit

* **New Features**
  * Enhanced cross-dimension backend transfers by tracking and clearing client-visible state (entities, boss bars, player list entries, and scoreboard objectives).
* **Bug Fixes**
  * Improved world reset behavior to stop all sounds, end rain/thunder effects, apply difficulty changes, and restore player pitch/yaw.
  * Strengthened entity targeting/ID rewriting to avoid collisions and suppress unintended backend-originated self-spawns.
* **Tests**
  * Expanded integration coverage for transfer clearing, runtime ID rewrite edge cases, and backend state tracker behavior.

### CODE REVIEW

File: integration/proxy/proxy.go [hunk 1]
```diff
@@ -409,64 +431,68 @@ func (s *session) rewriteClientPacket(pk packet.Packet) {
...
-			pk.ContainerEntityUniqueID = s.clientUniqueID
-		}
+		rewriteUniqueID(&pk.ContainerEntityUniqueID)
 	case *packet.RemoveActor:
-		if pk.EntityUniqueID == s.player.UniqueId {
-			pk.EntityUniqueID = s.clientUniqueID
-		}
+		rewriteUniqueID(&pk.EntityUniqueID)
 	}
+	return true
```
[the remark this answers is not in the document]
Comment (HashimTheArab) [reply integration/proxy/proxy.go:500]: Fixed in c86550c. AddPainting now passes both identity fields through the same runtime/unique collision mapper, with a regression covering both fields.
